pkg/maps/lbmap/lbmap.go

```diff
@@ -152,23 +152,32 @@ func (lbmap *LBBPFMap) UpsertService(p *UpsertServiceParams) error {
 
 // UpsertMaglevLookupTable calculates Maglev lookup table for given backends, and
 // inserts into the Maglev BPF map.
-func (lbmap *LBBPFMap) UpsertMaglevLookupTable(svcID uint16, backends map[string]loadbalancer.BackendID, ipv6 bool) error {
+func (lbmap *LBBPFMap) UpsertMaglevLookupTable(svcID uint16, backends map[string]*loadbalancer.Backend, ipv6 bool) error {
 	backendNames := make([]string, 0, len(backends))
-	for name := range backends {
-		backendNames = append(backendNames, name)
+	for name, b := range backends {
+		if b != nil && b.Weight > 0 {
+			backendNames = append(backendNames, name)
+		}
 	}
 	// Maglev algorithm might produce different lookup table for the same
 	// set of backends listed in a different order. To avoid that sort
 	// backends by name, as the names are the same on all nodes (in opposite
 	// to backend IDs which are node-local).
-	sort.Strings(backendNames)
-	table := maglev.GetLookupTable(backendNames, lbmap.maglevTableSize)
-	for i, pos := range table {
-		lbmap.maglevBackendIDsBuffer[i] = backends[backendNames[pos]]
-	}
-
-	if err := updateMaglevTable(ipv6, svcID, lbmap.maglevBackendIDsBuffer); err != nil {
-		return err
+	// In case all backends are put into maintenance we delete the maglev table.
+	if len(backendNames) > 0 {
+		sort.Strings(backendNames)
+		table := maglev.GetLookupTable(backendNames, lbmap.maglevTableSize)
+		// In case there are no backends we should delete maglevBackendIDsBuffer
```

Comment:
Mind elaborating on this? The buffer is global, i.e., it's used by all services. I guess the comment might be misleading.

Reply:
you are so right, sorry that comment is not supposed to be here :(